plugins/hls-class-plugin/test/Main.hs

```diff
@@ -124,7 +124,7 @@ codeLensTests = testGroup
     , testCase "Do not construct error action!, Ticket3942one" $ do
         runSessionWithServer def classPlugin testDataDir $ do
             doc <- openDoc "Ticket3942one.hs" "haskell"
-            _ <- waitForDiagnosticsFromSource doc (T.unpack sourceTypecheck)
```

Comment:
This function is rather weird. I don't know why it isn't just a slight variant of `waitForDiagnostics` that adds an extra predicate...